src/common/XChainIntegrator.sol

```diff
@@ -4,21 +4,15 @@ pragma solidity 0.8.16;
 import { ERC20 } from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
 import { Auth } from "./Auth.sol";
 import "../interfaces/MsgStructs.sol";
-import "../interfaces/postOffice/IPostOffice.sol";
+import "../interfaces/postOffice/IPostman.sol";
 
 abstract contract XChainIntegrator is Auth {
-	mapping(address => Vault) public depositedVaults;
-	address[] internal vaultList;
+	mapping(address => Vault) public addrBook;
+	mapping(uint256 => address) internal postmanAddr;
```

Comment:
storing postmanAddr separately might over complicate things a little bit?
maybe its better to store addresses directly in the Vault structs? I realize that this is a more optimal storage-wise.
I'll think more about this as we test stuff, no need to change yet.

Reply:
My idea is to be able to upgrade a postman more easily. If we store the address directly then we need to change in each vault the address of a postman if we ever update one of them (which seems likely to happen with layerZero)